cmd/fleetctl/scripts.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"errors"
+	"html/template"
+	"net/http"
+	"os"
+	"strings"
+	"unicode/utf8"
+
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/fleetdm/fleet/v4/server/service"
+	"github.com/urfave/cli/v2"
+)
+
+func runScriptCommand() *cli.Command {
+	return &cli.Command{
+		Name:      "run-script",
+		Aliases:   []string{"run_script"},
+		Usage:     `Run a script on one host.`,
+		UsageText: `fleetctl run-script [options]`,
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:     "script-path",
+				Usage:    "The path to the script.",
+				Required: true,
+			},
+			&cli.StringFlag{
+				Name:     "host",
+				Usage:    "A host, specified by hostname, UUID, osquery host ID, or node key.",
+				Required: true,
+			},
+			configFlag(),
+			contextFlag(),
+			debugFlag(),
+		},
+		Action: func(c *cli.Context) error {
+			client, err := clientFromCLI(c)
+			if err != nil {
+				return err
+			}
+
+			path := c.String("script-path")
+			if err := validateScriptPath(path); err != nil {
+				return err
+			}
+
+			ident := c.String("host")
+			h, err := client.HostByIdentifier(ident)
+			if err != nil {
+				var nfe service.NotFoundErr
+				if errors.As(err, &nfe) {
+					return errors.New(fleet.ErrRunScriptHostNotFound)
+				}
+				var sce service.StatusCodeErr
+				if errors.As(err, &sce) {
+					if sce.StatusCode() == http.StatusForbidden {
+						return errors.New(fleet.ErrRunScriptForbidden)
+					}
+				}
+				return err
```

Comment:
had to do something similar recently, seems kind of silly

Reply:
Yeah, at some point we need to revisit how we are translating errors for display to maybe save some busy work converting between API, CLI, UI, etc. and also maybe make it easier to handle copy changes in one place.